mesonbuild/compilers/swift.py

```diff
@@ -115,6 +116,21 @@ def get_include_args(self, path: str, is_system: bool) -> T.List[str]:
     def get_compile_only_args(self) -> T.List[str]:
         return ['-c']
 
+    def get_option_compile_args(self, target: build.BuildTarget, env: Environment, subproject: T.Optional[str] = None
+                                ) -> T.List[str]:
+        args: T.List[str] = []
+
+        # Pass C compiler args to swiftc, notably -std=...
+        try:
+            c_langs = ['objc', 'c']
+            c_lang = next(lang for lang in c_langs if lang in target.compilers)
```

Comment:
We have a `mesonlib.first()` helper that I think would simplify this:
```python
lang = mesonlib.first(target.compilers, lambda x: x in {'objc', 'c'})
if lang is not None:
    cc = target.compilers[lang]
    args.extend(...)
```
with no need for a try/except block

Reply:
Nice!

Reply:
The order is important here though, it should generally prefer objc/c++ when available, so kept the iteration the other way around.